tools/generate_hotkeys.py

```diff
@@ -13,34 +17,135 @@
 # Name of the script
 SCRIPT_NAME = PurePath(__file__).name
 
-categories = {
-    category: {
-        item["help_text"]: item["keys"]
-        for item in KEY_BINDINGS.values()
-        if item["key_category"] == category
+# Exclude keys from duplicate keys checking
+KEYS_TO_EXCLUDE = ["q", "e", "m", "r"]
+
+
+def main(check_only: bool = False) -> None:
+    if check_only:
+        lint_hotkeys_file()
+    else:
+        generate_hotkeys_file()
+
+
+def read_help_categories() -> Dict[str, Dict[str, List[str]]]:
+    """
+    Get all help categories from keys.py
+    """
+    categories = {
+        category: {
+            item["help_text"]: item["keys"]
+            for item in KEY_BINDINGS.values()
+            if item["key_category"] == category
+        }
+        for category in HELP_CATEGORIES.keys()
```

Comment:
What is the reasoning here for using the second `dict` inside the main `dict`?
Later `categories[action]` is anyway converted into a list of tuples using `items()`.
So why not just store it as a list of tuples in the first place?

Another point here is, you are looping through the large ~70 `KEY_BINDINGS` for every category (6). Instead why not try using a `defaultdict(list)` and run through `KEY_BINDINGS` only once?

Reply:
Thanks for pointing this out @Rohitth007, though I think this was in the original version from early 2021?

It's a good refactor to have, though being outside of the main application I'm not convinced the optimization is required :) However, it's good this filtering pattern, as this might be more warranted where we partition a larger data-structure there, eg. users based on presence.